File: backend/app/schemas/analytics.py
from pydantic import BaseModel
from app.schemas.common import ChartPoint, SeriesPoint, NamedCount


class AgentPerformance(BaseModel):
    agent: str
    executions: int
    successRate: float
    avgConfidence: float


class AnalyticsResponse(BaseModel):
    recommendationsOverTime: list[SeriesPoint]
    segmentDistribution: list[ChartPoint]
    countryDistribution: list[ChartPoint]
    businessRuleUsage: list[NamedCount]
    guardrailViolations: list[NamedCount]
    topAgents: list[AgentPerformance]

File: frontend/src/types/analytics.ts
import type { ChartPoint, NamedCount, SeriesPoint } from "./common"

export interface AgentPerformance {
  agent: string
  executions: number
  successRate: number
  avgConfidence: number
}

export interface AnalyticsResponse {
  recommendationsOverTime: SeriesPoint[]
  segmentDistribution: ChartPoint[]
  countryDistribution: ChartPoint[]
  businessRuleUsage: NamedCount[]
  guardrailViolations: NamedCount[]
  topAgents: AgentPerformance[]
}
